src/grammar/formula.rs

```diff
@@ -869,7 +869,7 @@ impl Formula {
             }) => Formula::Terminal(FormulaTerminal::Log2FoldChange {
                 sample_a: sample_a.into(),
                 sample_b: sample_b.into(),
-                predicate: !*predicate,
+                predicate: predicate.clone(),
```

Comment:
Thanks for catching the double negation (`negate` + `apply_negations`). As clippy suggests, since l2fcp is Copy, we could just write
```suggestion
                predicate: *predicate,
```
instead